### DISCUSSION TITLE
(Answer: Can't be changed through extensions) Controlling PDF background color (behind pages) in Edge / Chromium

### DISCUSSION THREAD

**Discussion by matteyas** [darkreader/darkreader#7501] [Q&A] [answered]
I've noticed that the extension, if turned on / off on a PDF, switches the background color (behind the actual PDF pages). I think it switches between #333333 and #c4c4c4. Is it possible to control this manually? I tried searching for those colors in the dev tools customization page but didn't find anything that seemed connected to the PDF reader mode.

Picture clarification, with Dark Reader on in the first (lighter gray behind the pages) and off in the second (darker gray):
![image](https://user-images.githubusercontent.com/12509132/143891909-5fbd467c-9811-4adc-8839-4e27c0b025a9.png)
![image](https://user-images.githubusercontent.com/12509132/143891994-712a9245-9a1a-4b94-96ef-11303321d422.png)

**Comment by Gusted** [comment]
Due to technical difficulties, that's not possible.

**Reply by matteyas** [reply]
Thanks for the answer, I'll mark it if you can clarify a bit. There is a change when the extension is used, so something is going on. Is it a non-local change? I'd be fine with changing it even if it changes the color in other places. Is it not controlled by the extension at all?

**Reply by Gusted** [reply]
Imagine you have a canvas with a beautiful painting on it, you cannot draw or touch individual things on that canvas. But you can apply this filter over the whole canvas. This filter doesn't map to a specific color, but to the inverse of the color, it's not controlled to what the output would be.

**Reply by matteyas** [reply]
Unfortunate! Thanks for the help.

**Comment by luo-chuan** [comment, marked as answer]
If you use Firefox, you can edit [userContent.css](https://davidwalsh.name/firefox-user-stylesheet) like this:
```css
@media (prefers-color-scheme: dark){
  .pdfViewer{filter: invert(1) hue-rotate(180deg);}
}
```
If you use [Chrome](https://chromewebstore.google.com/detail/pdf-reader/ieepebpjnkhaiioojkepfniodjmjjihl) or [Edge](https://microsoftedge.microsoft.com/addons/detail/pdf-reader/nhppiemcomgngbgdeffdgkhnkjlgpcdi), install the extention "PDF Reader", then edit the extention option like this:
```css
html[data-theme*="dark"] .pdfViewer{
  filter: invert(1) hue-rotate(180deg);
}
```

**Reply by matteyas** [reply]
I think I did try PDF Reader but it didn't quite work if I recall correctly; I use Opera GX so it's not a pure Chromium browser.

I found another extension, PDF Viewer, which does work and although it doesn't support styling natively, it does render PDF:s in HTML so I wrote an AHK script to input some JS code into the console. (The PDF Viewer extension kinda takes ownership when opening PDF:s so my CSS/JS injection extension can't run.)

This is the script I use if anyone would possibly want to consider this solution (the result isn't perfect by any means, so any suggestions for JS code—a {raw} sendinput near the bottom of the script—that runs better visual tweaks would be appreciated):
```ahk
; dark mode PDF (operagx)
^+a::
  KeyWait, Shift
  KeyWait, Ctrl
  KeyWait, a

  WinGet, Active_ID, ID, A
  WinGet, procname, ProcessName, ahk_id %Active_ID%
  if (procname = "opera.exe") {
    WinGetActiveTitle, title
    ControlGet, dev_visible,Visible,, Chrome_RenderWidgetHostHWND2, %title%

    ; if it's not enabled already, send ctrl+shift+i to enter dev / inspect mode
    if (dev_visible = 0) {
      send +^i
    }

    ; wait until it's active
    while (dev_visible = 0) {
      WinGetActiveTitle, title
      ControlGet, dev_visible,Visible,, Chrome_RenderWidgetHostHWND2, %title%
      Sleep, 250
    }

    ; run some javascript to modify elements, then exit dev mode
    sendinput {RAW}elems = document.getElementsByClassName("pdfViewer")[0].children; for (elem of elems) elem.style.webkitFilter = 'brightness(0.99) invert(0.9)';
    sleep, 500
    send {Enter}
    sleep, 500
    send +^i
  }
  return
```

Before vs. After:
![image](https://github.com/darkreader/darkreader/assets/12509132/999ff135-83b8-4805-8ee8-c0e118297e77)
![image](https://github.com/darkreader/darkreader/assets/12509132/6d0fadde-e50a-4c4b-8f88-2712dfbf77bd)

**Reply by matteyas** [reply]
To verify, and add another comparison shot, I tried the PDF Reader in Edge using the suggested styling:
![image](https://github.com/darkreader/darkreader/assets/12509132/aeb8750c-9a68-4eba-9b12-8bcf885aec05)

**Comment by Artectdis** [comment]
For people years later looking for how to control the PDF background color beside the pages in Chromium-based browsers, it's technically possible with another method.

I have created **PDF Dark Overlay for Chrome PDF Viewer** for this. This uses Tampermonkey (although you can do it temporarily on the page); for easy repetition the next time you want to activate a clean dark mode together with the background.

Adds sidebar masks and backdrop-filter over Chrome's PDF viewer, allowing tools like Dark Reader to have a clean dark PDF viewer without:
1. recompiling and changing the background color every update,
2. filtering only all colors unselectively,
3. page resizing until the background disappears,
4. transferring to a different PDF viewer, often requiring uploads for every file you look at, unlike the built-in Chrome viewer.

**Drag the overlay edges to resize them or use the floating menu to adjust brightness and offset.**

<img width="1365" height="767" alt="Screenshot 2026-07-27 135031" src="https://github.com/user-attachments/assets/1cc7b6e4-4c6a-4150-a682-2212792fab49" />
FIRST EXAMPLE (note, brightness can be tweaked, settings menu can be closed, also drag the sidebar's edges to adjust their width depending on magnification to hide the white background)

<img width="1365" height="767" alt="Screenshot 2026-07-27 134922" src="https://github.com/user-attachments/assets/0eafeb5a-c945-4c2e-ae61-e8b2bce8b935" />
FULL DARK EXAMPLE (topbar brightness and sidebar brightness all the way down)
<hr/>
Tweaks automatically save per user, and defaults can be changed in the variables section.


_Requires Allow access to file URLs to reach file:/// or objects in files, as well as the Allow User Scripts so that Tampermonkey can run._


**USERSCRIPT LINK (get the Tampermonkey Chrome extension if you do not have it and add the requirements to run on files):**
https://greasyfork.org/en/scripts/588718-pdf-native-viewer-dark-overlay

GITHUB (if you want to see the code or full description): https://github.com/Artectdis/pdf-dark-overlay

**Reply by alexanderby** [reply]
That's a very nice idea! Do you have to drag the overlays manually?
